### PR TITLE
ant: 1.10.11 -> 1.10.15, add update script

### LINKED ISSUE #275640: upgrade apache-ant to fix SOURCE_DATE_EPOCH bug
Ant versions prior to 1.10.13 might fail to parse the `SOURCE_DATE_EPOCH` environment variable, setting timestamps to a wrong value (see [ant/WHATSNEW](https://github.com/apache/ant/blob/master/WHATSNEW) and [Fix integer overflow when parsing SOURCE_DATE_EPOCH by mizdebsk · Pull Request #186 · apache/ant](https://github.com/apache/ant/pull/186)). This is problematic in Nix since it sets that variable for reproducibility.

### PR DESCRIPTION
**Pull request by tomodachi94**
Changelogs: https://downloads.apache.org/ant/

Closes #275640

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/build-managers/apache-ant/default.nix [hunk 1]
```diff
@@ -78,6 +78,10 @@ stdenv.mkDerivation rec {
       chmod +x $out/bin/ant
     ''; # */
 
+  passthru = {
+    updateScript = ./update.sh;
```
Comment (NickCao) [inline pkgs/development/tools/build-managers/apache-ant/default.nix:82]: Does `gitUpdater` work for this?

Comment (tomodachi94) [reply pkgs/development/tools/build-managers/apache-ant/default.nix]: Indeed; I forgot about `gitUpdater` :sweat_smile: Used in latest push.

**Review by NickCao — APPROVED** [review]

**Comment by NickCao** [comment]
Running nixpkgs-review (the number of rebuilds is kind of large)

**Comment by tomodachi94** [comment]
Beware, you're rebuilding LibreOffice several times (due to the `ant <- rhino <- libreoffice-*fresh*` chain.

**Comment by NickCao** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353301`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 8 packages marked as broken and skipped:</summary>
  <ul>
    <li>hdfview</li>
    <li>python311Packages.worldengine</li>
    <li>python311Packages.worldengine.dist</li>
    <li>python312Packages.worldengine</li>
    <li>python312Packages.worldengine.dist</li>
    <li>spring</li>
    <li>springLobby</li>
    <li>vuze</li>
  </ul>
</details>
<details>
  <summary>:x: 29 packages failed to build:</summary>
  <ul>
    <li>hap-py</li>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.osmnx</li>
    <li>python311Packages.osmnx.dist</li>
    <li>python311Packages.python-mapnik</li>
    <li>python311Packages.python-mapnik.dist</li>
    <li>python311Packages.rasterio</li>
    <li>python311Packages.rasterio.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python311Packages.rioxarray</li>
    <li>python311Packages.rioxarray.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.osmnx</li>
    <li>python312Packages.osmnx.dist</li>
    <li>python312Packages.python-mapnik</li>
    <li>python312Packages.python-mapnik.dist</li>
    <li>python312Packages.rasterio</li>
    <li>python312Packages.rasterio.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
    <li>python312Packages.rioxarray</li>
    <li>python312Packages.rioxarray.dist</li>
    <li>python312Packages.textnets</li>
    <li>python312Packages.textnets.dist</li>
    <li>t-rex</li>
    <li>xtreemfs</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 285 packages built:</summary>
  <ul>
    <li>DisnixWebService</li>
    <li>abcl</li>
    <li>ant</li>
    <li>apacheHttpdPackages.mod_tile</li>
    <li>arduino</li>
    <li>arduino-core</li>
    <li>arduino-core-unwrapped</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>asciidoc-full-with-plugins.dist</li>
    <li>asciidoc-full.dist</li>
    <li>autopsy</li>
    <li>axis2</li>
    <li>batik</li>
    <li>brmodelo</li>
    <li>buck</li>
    <li>calcoo</li>
    <li>checkov</li>
    <li>checkov.dist</li>
    <li>clevis</li>
    <li>clevis.man</li>
    <li>cloudcompare</li>
    <li>collabora-online</li>
    <li>dataexplorer</li>
    <li>dayon</li>
    <li>disorderfs</li>
    <li>domination</li>
    <li>entwine</li>
    <li>fop</li>
    <li>freeipa</li>
    <li>freemind</li>
    <li>freetts</li>
    <li>gdal (python312Packages.gdal)</li>
    <li>gephi</li>
    <li>gmt</li>
    <li>gogui</li>
    <li>gotenberg</li>
    <li>gplates</li>
    <li>gpsbabel-gui</li>
    <li>gpsbabel-gui.doc</li>
    <li>grass</li>
    <li>hal-hardware-analyzer</li>
    <li>hexgui</li>
    <li>hydraAntLogger</li>
    <li>i2p</li>
    <li>igraph</li>
    <li>igraph.dev</li>
    <li>igraph.doc</li>
    <li>ili2c</li>
    <li>jailer</li>
    <li>jameica</li>
    <li>jasmin</li>
    <li>java-service-wrapper</li>
    <li>javaCup</li>
    <li>javacc</li>
    <li>jdepend</li>
    <li>jedit</li>
    <li>jetbrains.aqua</li>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.jcef</li>
    <li>jetbrains.jcef.unpacked</li>
    <li>jetbrains.jdk</li>
    <li>jetbrains.jdk.debug</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
    <li>jetbrains.writerside</li>
    <li>jffi</li>
    <li>jing</li>
    <li>jitsi</li>
    <li>jna</li>
    <li>jogl</li>
    <li>jpsxdec</li>
    <li>kakounePlugins.rep</li>
    <li>libleidenalg</li>
    <li>libreoffice (libreoffice-still)</li>
    <li>libreoffice-collabora</li>
    <li>libreoffice-fresh</li>
    <li>libreoffice-fresh-unwrapped</li>
    <li>libreoffice-qt (libreoffice-qt-still)</li>
    <li>libreoffice-qt-fresh</li>
    <li>libreoffice-qt-fresh-unwrapped</li>
    <li>libreoffice-qt-still-unwrapped</li>
    <li>libreoffice-qt6 (libreoffice-qt6-still)</li>
    <li>libreoffice-qt6-fresh</li>
    <li>libreoffice-qt6-fresh-unwrapped</li>
    <li>libreoffice-qt6-unwrapped</li>
    <li>libreoffice-still-unwrapped</li>
    <li>liquibase</li>
    <li>lomiri.lomiri-docviewer-app</li>
    <li>luksmeta</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapproxy.dist</li>
    <li>mapserver</li>
    <li>merkaartor</li>
    <li>mindustry</li>
    <li>mindustry-wayland</li>
    <li>mkgmap</li>
    <li>mkgmap-splitter</li>
    <li>mysql-workbench</li>
    <li>mysql_jdbc</li>
    <li>octavePackages.arduino</li>
    <li>octavePackages.mapping</li>
    <li>openorienteering-mapper</li>
    <li>openrocket</li>
    <li>paperwork</li>
    <li>paperwork.dist</li>
    <li>paraview</li>
    <li>pattypan</li>
    <li>pdal</li>
    <li>perl538Packages.Tirex</li>
    <li>perl538Packages.Tirex.devdoc</li>
    <li>perl540Packages.Tirex</li>
    <li>perl540Packages.Tirex.devdoc</li>
    <li>polymake</li>
    <li>processing</li>
    <li>projectlibre</li>
    <li>python311Packages.bsuite</li>
    <li>python311Packages.bsuite.dist</li>
    <li>python311Packages.cartopy</li>
    <li>python311Packages.cartopy.dist</li>
    <li>python311Packages.django-bootstrap4</li>
    <li>python311Packages.django-bootstrap4.dist</li>
    <li>python311Packages.django-bootstrap5</li>
    <li>python311Packages.django-bootstrap5.dist</li>
    <li>python311Packages.drf-extra-fields</li>
    <li>python311Packages.drf-extra-fields.dist</li>
    <li>python311Packages.explorerscript</li>
    <li>python311Packages.explorerscript.dist</li>
    <li>python311Packages.fiona</li>
    <li>python311Packages.fiona.dist</li>
    <li>python311Packages.folium</li>
    <li>python311Packages.folium.dist</li>
    <li>python311Packages.gdal</li>
    <li>python311Packages.geodatasets</li>
    <li>python311Packages.geodatasets.dist</li>
    <li>python311Packages.geopandas</li>
    <li>python311Packages.geopandas.dist</li>
    <li>python311Packages.geoparquet</li>
    <li>python311Packages.geoparquet.dist</li>
    <li>python311Packages.igraph</li>
    <li>python311Packages.igraph.dist</li>
    <li>python311Packages.inequality</li>
    <li>python311Packages.inequality.dist</li>
    <li>python311Packages.kmapper</li>
    <li>python311Packages.kmapper.dist</li>
    <li>python311Packages.leidenalg</li>
    <li>python311Packages.leidenalg.dist</li>
    <li>python311Packages.libpysal</li>
    <li>python311Packages.libpysal.dist</li>
    <li>python311Packages.mapclassify</li>
    <li>python311Packages.mapclassify.dist</li>
    <li>python311Packages.momepy</li>
    <li>python311Packages.momepy.dist</li>
    <li>python311Packages.netbox-documents</li>
    <li>python311Packages.netbox-documents.dist</li>
    <li>python311Packages.paperwork-backend</li>
    <li>python311Packages.paperwork-backend.dist</li>
    <li>python311Packages.paperwork-shell</li>
    <li>python311Packages.paperwork-shell.dist</li>
    <li>python311Packages.plotnine</li>
    <li>python311Packages.plotnine.dist</li>
    <li>python311Packages.pygmt</li>
    <li>python311Packages.pygmt.dist</li>
    <li>python311Packages.pyogrio</li>
    <li>python311Packages.pyogrio.dist</li>
    <li>python311Packages.scikit-tda</li>
    <li>python311Packages.scikit-tda.dist</li>
    <li>python311Packages.shimmy</li>
    <li>python311Packages.shimmy.dist</li>
    <li>python311Packages.skytemple-dtef</li>
    <li>python311Packages.skytemple-dtef.dist</li>
    <li>python311Packages.skytemple-files</li>
    <li>python311Packages.skytemple-files.dist</li>
    <li>python311Packages.skytemple-ssb-debugger</li>
    <li>python311Packages.skytemple-ssb-debugger.dist</li>
    <li>python311Packages.textnets</li>
    <li>python311Packages.textnets.dist</li>
    <li>python311Packages.wktutils</li>
    <li>python311Packages.wktutils.dist</li>
    <li>python312Packages.bsuite</li>
    <li>python312Packages.bsuite.dist</li>
    <li>python312Packages.cartopy</li>
    <li>python312Packages.cartopy.dist</li>
    <li>python312Packages.django-bootstrap4</li>
    <li>python312Packages.django-bootstrap4.dist</li>
    <li>python312Packages.django-bootstrap5</li>
    <li>python312Packages.django-bootstrap5.dist</li>
    <li>python312Packages.drf-extra-fields</li>
    <li>python312Packages.drf-extra-fields.dist</li>
    <li>python312Packages.explorerscript</li>
    <li>python312Packages.explorerscript.dist</li>
    <li>python312Packages.fiona</li>
    <li>python312Packages.fiona.dist</li>
    <li>python312Packages.folium</li>
    <li>python312Packages.folium.dist</li>
    <li>python312Packages.geodatasets</li>
    <li>python312Packages.geodatasets.dist</li>
    <li>python312Packages.geopandas</li>
    <li>python312Packages.geopandas.dist</li>
    <li>python312Packages.geoparquet</li>
    <li>python312Packages.geoparquet.dist</li>
    <li>python312Packages.igraph</li>
    <li>python312Packages.igraph.dist</li>
    <li>python312Packages.inequality</li>
    <li>python312Packages.inequality.dist</li>
    <li>python312Packages.kmapper</li>
    <li>python312Packages.kmapper.dist</li>
    <li>python312Packages.leidenalg</li>
    <li>python312Packages.leidenalg.dist</li>
    <li>python312Packages.libpysal</li>
    <li>python312Packages.libpysal.dist</li>
    <li>python312Packages.mapclassify</li>
    <li>python312Packages.mapclassify.dist</li>
    <li>python312Packages.momepy</li>
    <li>python312Packages.momepy.dist</li>
    <li>python312Packages.netbox-documents</li>
    <li>python312Packages.netbox-documents.dist</li>
    <li>python312Packages.paperwork-backend</li>
    <li>python312Packages.paperwork-backend.dist</li>
    <li>python312Packages.paperwork-shell</li>
    <li>python312Packages.paperwork-shell.dist</li>
    <li>python312Packages.plotnine</li>
    <li>python312Packages.plotnine.dist</li>
    <li>python312Packages.pygmt</li>
    <li>python312Packages.pygmt.dist</li>
    <li>python312Packages.pyogrio</li>
    <li>python312Packages.pyogrio.dist</li>
    <li>python312Packages.scikit-tda</li>
    <li>python312Packages.scikit-tda.dist</li>
    <li>python312Packages.skytemple-dtef</li>
    <li>python312Packages.skytemple-dtef.dist</li>
    <li>python312Packages.skytemple-files</li>
    <li>python312Packages.skytemple-files.dist</li>
    <li>python312Packages.skytemple-ssb-debugger</li>
    <li>python312Packages.skytemple-ssb-debugger.dist</li>
    <li>python312Packages.wktutils</li>
    <li>python312Packages.wktutils.dist</li>
    <li>pytrainer</li>
    <li>pytrainer.dist</li>
    <li>qgis</li>
    <li>qgis-ltr</li>
    <li>qmapshack</li>
    <li>rhino</li>
    <li>rstudio</li>
    <li>rstudio-server</li>
    <li>rstudioServerWrapper</li>
    <li>rstudioWrapper</li>
    <li>rtg-tools</li>
    <li>saga</li>
    <li>skytemple</li>
    <li>skytemple.dist</li>
    <li>sleuthkit</li>
    <li>squirrel-sql</li>
    <li>sumo</li>
    <li>survex</li>
    <li>sweethome3d.application</li>
    <li>sweethome3d.furniture-editor</li>
    <li>sweethome3d.textures-editor</li>
    <li>tang</li>
    <li>tang.man</li>
    <li>teensyduino</li>
    <li>therion</li>
    <li>trimmomatic</li>
    <li>tunnelx</li>
    <li>tvbrowser</li>
    <li>udig</li>
    <li>unoconv</li>
    <li>vpv</li>
    <li>xmloscopy</li>
  </ul>
</details>

**Comment by tomodachi94** [comment]
Investigating the failures.

**Comment by tomodachi94** [comment]
The following packages were failing before this PR:
* [hap-py](https://hydra.nixos.org/build/276695333)
* [xtreemfs](https://hydra.nixos.org/build/276807334)
* [t-rex](https://hydra.nixos.org/build/276727249)
* python3Packages...
    * [morecantile](https://hydra.nixos.org/build/276845779) (depends on broken rasterio)
    * [osmnx](https://hydra.nixos.org/build/276908008) (rasterio)
    * [python-mapnik](https://hydra.nixos.org/build/277145787)
    * [rasterio](https://hydra.nixos.org/build/276880377)
    * [rio-tiler](https://hydra.nixos.org/build/276895214) (rasterio (I'm starting to see a pattern...))
    * [rioxarray](https://hydra.nixos.org/build/276836737)

Currently building `python312Packages.textnets` locally to determine why it failed.

**Comment by tomodachi94** [comment]
`python312Packages.textnets` no longer fails when I rebase onto `master`. It also [used to fail on Hydra](https://hydra.nixos.org/build/277272787) until [an hour ago](https://hydra.nixos.org/build/277279206).

**Comment by tomodachi94** [comment]
@ofborg build ant python312Packages.textnets python312Packages.textnets.dist

**Review by Infinidoge — APPROVED** [review]
Looks good to me! `gitUpdater` is very convenient. Given everything builds, should be good to merge :)

**Comment by tomodachi94** [comment]
Is this okay to merge?
* nixpkgs-review by NickCao early in this PR reveals no new breakages
* Has 1 approval. The other approval was for identical content; the force-push was a rebase onto master to get rid of a failure

**Comment by tomodachi94** [comment]
Oops, this definitely should have gone to staging (nixpkgs-review on another PR that changes Ant shows ~800 packages needing to be rebuilt, mostly indirect dependencies). I'll be more careful in the future.

**Pull request merged**